feat: Add delivery list view

Ronald752 opened the pull request:
# Description

In this PR, the views corresponding to the food deliveries are being developed, which are the list, the delivery creation and the details of each delivery.

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# Screenshot:

At the moment I'm using mockapi and the date gives it to me that way, but when connecting to the backend, the date will be displayed with the format set in the mockups.

![image](https://github.com/ISPP-07/frontend-cyc/assets/73199700/e85d1929-4315-488a-80b7-7955270edffe)

![image](https://github.com/ISPP-07/frontend-cyc/assets/73199700/d8c0a128-9449-4713-9fde-35b33e6fced3)

![image](https://github.com/ISPP-07/frontend-cyc/assets/73199700/e7675671-f370-4ce2-bb2b-e70c712fac46)

![image](https://github.com/ISPP-07/frontend-cyc/assets/73199700/f59180d1-7a91-42cd-bce2-086b72307051)

Ronald752 commented:
At the moment my functionality is not complete, but advanced enough to request a review of what is already developed.
In the list of deliverables, a part of its details is missing.
Note that the date is not shown in the same format as the mockup, because I am waiting to know in which format it will be sent by the backend. 
Finally, I need to connect the delivery record so that it is created correctly.

sebasruii commented on src/app/components/sidebar.jsx:
When I press on `Añadir entregas`, the modal does not show up
![image](https://github.com/ISPP-07/frontend-cyc/assets/73228845/bbdae95c-011f-4bd0-a4d3-af34aa4886f1)

Ronald752 replied:
I have checked which performs the redirect to the view, I think the error is due to not passing the modal status correctly, also I noticed that when you open the modal, there is no specific url, it may vary depending on the background. 
This error happens with all the other view creation buttons.

sebasruii commented on src/app/deliveries/page.jsx:
Also (maybe this is not related to your implementation), when the page is displayed in a narrow viewport, the + (button for creating a delivery) does not pop up the modal
![image](https://github.com/ISPP-07/frontend-cyc/assets/73228845/ef553cb9-9e72-4e5f-b64e-504e8a85d480)

However, it does pop up in a wide viewport (such as full screen)

Ronald752 replied:
I just checked and you are right, however I noticed that in the family and inventory views, the create button does not work either when the screen is reduced in size. Possibly it has to do with some property of the search bar.

Ronald752 replied:
I've tried to fix it by using a grip with three columns, but I'm having trouble sizing the buttons in the last column and there's a lot of white space.

I have managed to get them aligned by applying a slight offset.

However I have detected that when applying the scrolling, this causes that when changing the size of the screen, everything looks messy, but the current version is responsive.
Therefore I have preferred to remove the scrolling until I find a solution that does not affect the change of screen dimensions.

Ronald752 replied:
Regarding the + button, I've fixed it, apparently they hadn't added the handleClick feature.

sebasruii reviewed (changes requested):
Great job Ronald! I really like your new designs. There is some styling that could use a little improvement:
![image](https://github.com/ISPP-07/frontend-cyc/assets/73228845/89179083-09ef-425c-86cf-ece978fe6f47)
The input fields should be aligned in their respective column. Overall great job, and Good use of MockAPI since backend is not ready yet. Check the reviews on these files:

Ronald752 commented on src/app/deliveries/page.jsx:
I have already made the change to icons, I hope you like it.

Ronald752 commented:
At the moment I have added the functionality that when the status of a delivery is changed, a PUT request is made.

sebasruii commented on src/app/components/DeliveriesForm.jsx:
Add a `event.preventDefault()` inside the form submit handler (pass event as a parameter) to avoid this 
![image](https://github.com/ISPP-07/frontend-cyc/assets/73228845/fcd11e65-12d0-4135-9963-836509f50ac5)

sebasruii replied:
Please, remember to remove the console.log()
![image](https://github.com/ISPP-07/frontend-cyc/assets/73228845/e68bf682-455a-4cac-acea-b84eb7ae8b49)

sebasruii replied:
(Just remove the ones you use for this feature)

sebasruii reviewed (changes requested):
Nice changes, please resolve conflicts and take into account the following changes:

sebasruii commented:
It is also necessary that the update method is connected to the backend and not MockAPI 
![image](https://github.com/ISPP-07/frontend-cyc/assets/73228845/533bf77a-259e-4010-916b-950708845e06)
The edit view is not currently implement, only the update of delivery status

Ronald752 commented:
I have made the changes you requested, I have implemented the functionality that allows you to change the status of a delivery. I also modified the buttons to keep the styles of the other views, however the edit button does nothing at the moment.

sebasruii reviewed (approved):
Great job Ronald 👍🏻
